### PR TITLE
PN-326: As a user, My Matches table should not show matches that don't involve my cases

### PR DESCRIPTION
**Pull request by allasm**

### CODE REVIEW

File: matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js [hunk 1]
```diff
@@ -267,12 +275,22 @@ var PhenoTips = (function (PhenoTips) {
             var hasGeneExomeTypeMatch = match.matchingGenesTypes.indexOf(this._filterValues.geneStatus) > -1;
             // by match status (rejected, saved, uncategorized)
             var hasMatchStatusMatch = this._filterValues.matchStatus[match.status];
+
+            var recordOwnershipMatch = function(record) {
+                                         return (this._filterValues.ownerStatus["me"] && record.ownership.userIsOwner)
+                                             || (this._filterValues.ownerStatus["group"] && record.ownership.userGroupIsOwner)
+                                             || (this._filterValues.ownerStatus["public"] && record.ownership.publicRecord)
+                                       };
+            var referenceRecordMatch = match.reference.serverid == "local" && recordOwnershipMatch(match.reference);
```
Comment (veronikaslc) [inline matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js]: couple problems: 
- not `serverid`, its `serverId`
- for local matches we leave `serverId` just blank `""`

File: matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js [hunk 2]
```diff
@@ -267,12 +275,22 @@ var PhenoTips = (function (PhenoTips) {
             var hasGeneExomeTypeMatch = match.matchingGenesTypes.indexOf(this._filterValues.geneStatus) > -1;
             // by match status (rejected, saved, uncategorized)
             var hasMatchStatusMatch = this._filterValues.matchStatus[match.status];
+
+            var recordOwnershipMatch = function(record) {
+                                         return (this._filterValues.ownerStatus["me"] && record.ownership.userIsOwner)
+                                             || (this._filterValues.ownerStatus["group"] && record.ownership.userGroupIsOwner)
+                                             || (this._filterValues.ownerStatus["public"] && record.ownership.publicRecord)
+                                       };
```
Comment (veronikaslc) [inline matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js]: `.bind(this)`

File: matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js [hunk 3]
```diff
@@ -267,12 +275,22 @@ var PhenoTips = (function (PhenoTips) {
...
+            var referenceRecordMatch = match.reference.serverid == "local" && recordOwnershipMatch(match.reference);
+            var matchedRecordMatch = match.matched.serverid == "local" && recordOwnershipMatch(match.matched);
+            var hasOwnershipMatch = referenceRecordMatch || matchedRecordMatch || this._filterValues.ownerStatus["others"];
```
Comment (veronikaslc) [inline matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js]: By this logic `others` equivalent to `everything`. If `others` checkbox is selected all cases will be shown even if all other checkboxes are unselected.

**Comment by allasm** [comment]
Closed in favour of #194

**Pull request closed without merging**